data-prepper-plugins/kafka-connect-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafkaconnect/source/KafkaConnectSource.java

```diff
@@ -0,0 +1,101 @@
+ */
+
+package org.opensearch.dataprepper.plugins.kafkaconnect.source;
+
+import org.apache.kafka.connect.runtime.WorkerConfig;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.configuration.PipelineDescription;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.Source;
+import org.opensearch.dataprepper.plugins.kafka.extension.KafkaClusterConfigSupplier;
+import org.opensearch.dataprepper.plugins.kafka.util.KafkaSourceSecurityConfigurer;
+import org.opensearch.dataprepper.plugins.kafkaconnect.configuration.ConnectorConfig;
+import org.opensearch.dataprepper.plugins.kafkaconnect.extension.KafkaConnectConfig;
+import org.opensearch.dataprepper.plugins.kafkaconnect.extension.KafkaConnectConfigSupplier;
+import org.opensearch.dataprepper.plugins.kafkaconnect.extension.WorkerProperties;
+import org.opensearch.dataprepper.plugins.kafkaconnect.util.Connector;
+import org.opensearch.dataprepper.plugins.kafkaconnect.util.KafkaConnect;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * The abstraction of the kafka connect source.
+ * The kafka connect and connectors are configured and runs async here.
+ */
+@SuppressWarnings("deprecation")
+public abstract class KafkaConnectSource implements Source<Record<Object>> {
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaConnectSource.class);
+    private final ConnectorConfig connectorConfig;
+    private final KafkaConnectConfig kafkaConnectConfig;
+    private final String pipelineName;
+    private KafkaConnect kafkaConnect;
+
+    public KafkaConnectSource(final ConnectorConfig connectorConfig,
+                              final PluginMetrics pluginMetrics,
+                              final PipelineDescription pipelineDescription,
+                              final KafkaClusterConfigSupplier kafkaClusterConfigSupplier,
+                              final KafkaConnectConfigSupplier kafkaConnectConfigSupplier) {
+        if (kafkaClusterConfigSupplier == null || kafkaConnectConfigSupplier == null) {
+            throw new IllegalArgumentException("Extensions: KafkaClusterConfig and KafkaConnectConfig cannot be null");
+        }
+        this.connectorConfig = connectorConfig;
+        this.pipelineName = pipelineDescription.getPipelineName();
+        this.kafkaConnectConfig = kafkaConnectConfigSupplier.getConfig();
+        this.updateConfig(kafkaClusterConfigSupplier);
+        this.kafkaConnect = KafkaConnect.getPipelineInstance(
+                pipelineName,
+                pluginMetrics,
+                kafkaConnectConfig.getConnectStartTimeout(),
+                kafkaConnectConfig.getConnectorStartTimeout());
+    }
+
+    @Override
+    public void start(Buffer<Record<Object>> buffer) {
```

Comment:
The `Buffer` isn't used. I understand that this will use the Kafka buffer. But, there should be some tighter connection here. What we don't want is for users to have a Data Prepper buffer that is different from the Kafka Connect topic.

Reply:
We can solve that in documentation/guidance. In OSDP, customers can use their own Kafka as buffer where the topic is filled in by customers. In OSI, we have shared MSK, and the topic is unique. (it is in the design doc)